src/main/java/com/example/commercetoolEx/controller/CategoryController.java

```diff
@@ -0,0 +1,42 @@
+package com.example.commercetoolEx.controller;
+
+import com.commercetools.api.models.category.Category;
+import com.commercetools.api.models.category.CategoryPagedQueryResponse;
+import com.example.commercetoolEx.dto.category.AddCategoryRequest;
+import com.example.commercetoolEx.dto.category.UpdateCategoryRequest;
+import com.example.commercetoolEx.service.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+import reactor.core.publisher.Mono;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(path = "/category")
+public class CategoryController {
+    private final CategoryService categoryService;
+
+    @GetMapping(path = "/all")
+    public Mono<CategoryPagedQueryResponse> getAllCategory() {
+        return categoryService.getAllCategory();
+    }
+
+    @GetMapping(path = "/{id}")
+    public Mono<Category> getCategoryById(@PathVariable String id) {
+        return categoryService.getCategoryById(id);
+    }
+
+    @PostMapping(path = "/add")
```

Comment:
the same thing apply to POST, if you want to create a resource, just post to `/`. Please change this to `@PostMapping` and it's enough